src/ngraph_builder.cc

```diff
@@ -4273,14 +4273,60 @@ static Status TranslateSelectOp(
   TF_RETURN_IF_ERROR(
       GetInputNodes(ng_op_map, op, &ng_input1, &ng_input2, &ng_input3));
 
-  // broadcast to make all tensors same shape, as required by ngraph select op
-  std::tie(ng_input1, ng_input2) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input1, ng_input2));
+  if (ng_input2->get_shape() != ng_input3->get_shape()) {
+    return errors::InvalidArgument(
+        "Input tensors 2 and 3 should have same shape");
+  }
+
+  auto ng_input1_shape = ng_input1->get_shape();
+  auto ng_input2_shape = ng_input2->get_shape();
+
+  auto ng_input1_rank = ng_input1->get_shape().size();
+  auto ng_input2_rank = ng_input2->get_shape().size();
+
+  if (!((ng_input1_shape == ng_input2_shape) ||
+        ((ng_input1_rank == 1) && (ng_input2_rank > ng_input1_rank) &&
+         (ng_input2_shape[0] == ng_input1_shape[0])))) {
+    return errors::InvalidArgument(
+        "Input tensor may have the same shape as condition. If condition is "
+        "rank 1, input may have higher rank, but its first dimension must "
+        "match the size of condition.");
+  }
+
+  int length = 0;
+  shared_ptr<ng::Node> ng_input_new, ng_select;
+  ng::AxisVector ng_axis_order;
+
+  for (size_t i = 0; i < ng_input2_rank; i++) {
+    if (ng_input1_shape[0] == ng_input2_shape[i]) {
+      length = ng_input2_rank - i - ng_input1_rank;
+    }
+  }
+
+  if (length != 0) {
+    // Condition tensor will be modified to align the condition tensor
+    // shape with input tensor shape index and fill the rest of the vector with
+    // 1s
+    // Eg: condition tensor [7], input tensor [7, 3, 2, 1]
```

Comment:
to make the example more generic, maybe we can say `[5, 7, 3, 2, 1]` . (to show that the realignment is not to make the sizes same, just to match up the first matching number)